File: sample/src/main/java/contacts/sample/cheatsheet/sim/java/QuerySimContactsActivity.java
package contacts.sample.cheatsheet.sim.java;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

import contacts.core.ContactsFactory;
import contacts.core.entities.SimContact;

public class QuerySimContactsActivity extends Activity {

    List<SimContact> getAllSimContacts() {
        return ContactsFactory.create(this).sim().query().find();
    }

    List<SimContact> getAllSimContactsWithPhoneNumber() {
        List<SimContact> simContacts = ContactsFactory.create(this).sim().query().find();
        List<SimContact> simContactsWithPhoneNumber = new ArrayList<>();
        for (SimContact simContact : simContacts) {
            if (simContact.getNumber() != null && !simContact.getNumber().isEmpty()) {
                simContactsWithPhoneNumber.add(simContact);
            }
        }
        return simContactsWithPhoneNumber;
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/sim/kotlin/QuerySimContactsActivity.kt
package contacts.sample.cheatsheet.sim.kotlin

import android.app.Activity
import contacts.core.Contacts
import contacts.core.entities.SimContact

class QuerySimContactsActivity : Activity() {

    fun getAllSimContacts(): List<SimContact> = Contacts(this).sim().query().find()

    fun getAllSimContactsWithPhoneNumber(): List<SimContact> = Contacts(this)
        .sim()
        .query()
        .find()
        .filter { !it.number.isNullOrEmpty() }
}